### ISSUE TITLE
extra '\r' in command output

### ISSUE THREAD

**Issue by jererc** [JasperVanDenBosch/fexpect#5]
Hello

is it possible to correct the output, which has extra '\r' at the end of lines than with the fabric commands ?

Jérôme

**Comment by JasperVanDenBosch** [comment]
Yes, I'll gladly look into this. Do you happen to have some example output and the corresponding script?

**Comment by jererc** [comment]
hey

http://pastebin.com/KCASAkHj

**Comment by JasperVanDenBosch** [comment]
Thank you! I think I can fix this tonight.

**Comment by jererc** [comment]
Actually, is it possible to get the same exact output than when running the command like fabric run/sudo/local methods (without the pexpect 'before' attr content and "Exiting fexpect for EOF.") and also the pexpect 'exit_status' attribute if needed ?

**Comment by JasperVanDenBosch** [comment]
Okay, I need a bit more time, perhaps tomorrow. Some of these things are pexpect internal, but we can workaround by changing the output as you did. I will create some verbosity settings and in the end it should be possible for it to create the same output as regular run/sudo/local.

**Comment by jererc** [comment]
Hello
I finally did my own lightweight expect using paramiko Channel, for speed purpose.
I'll check your work out.
Thanks

**Comment by jererc** [comment]
http://pastebin.com/WRRWf6jv

Feedbacks are welcome =)

**Comment by jererc** [comment]
Just created a repo there if you're interested or if you have suggestions:
https://github.com/jererc/sshex-py

Jerome

**Comment by JasperVanDenBosch** [comment]
I will definitely have a look. Which of the features you mentioned before would you still like to see in fexpect? Or will you switch to sshex-py entirely? Just so I know where to put my priorities :)

**Comment by jererc** [comment]
Well, i'm using my own project.
Although, i'm still interested in a fabric contrib with expect.

Thank you for the reactivity! =)
